src/gettxinfo.c

```diff
@@ -188,12 +189,29 @@ void gettxinfo(void) {
 	    handle_trx_bandswitch((int) freq);
 	}
 
+	/* read speed from rig */
+	if (cwkeyer == HAMLIB_KEYER) {
+	    value_t rig_cwspeed;
+	    retval = rig_get_level(my_rig, RIG_VFO_CURR, RIG_LEVEL_KEYSPD, &rig_cwspeed); /* initialize RIG_VFO_CURR */
+
+	    if (retval == RIG_OK) {
+		if (GetCWSpeed() != rig_cwspeed.i) { // FIXME: doesn't work if rig speed is between the values from CW_SPEEDS
+		    SetCWSpeed(rig_cwspeed.i);
```

Comment:
`SetCWSpeed` could be enhanced to accept any speed and select the closest available.

Reply:
That's basically that it is doing at the moment.

I still think tlf should drop this lookup-table-like string and simply accept what the rig sends - we can still have PgUp/Dn change the wpm in 2-increments. That would be effectively pretty much the same as before, plus allowing finer adjustements using the speed knob on the rig.